samza-core/src/main/java/org/apache/samza/execution/JobPlanner.java

```diff
@@ -82,10 +84,13 @@ ExecutionPlan getExecutionPlan(String runId) {
             userConfig.get(TaskConfig.INPUT_STREAMS()));
         allowedUserConfig.remove(TaskConfig.INPUT_STREAMS());
       }
-
       generatedConfig.putAll(getGeneratedConfig(runId));
     }
 
+    // Generate job.id and job.name configs from app.id and app.name if defined
+    MapConfig generatedJobConfigs = JobConfigUtil.generateJobIdAndName(new MapConfig(allowedUserConfig));
```

Comment:
this is not correct. The job name and id got generated for all the jobs here. This logic won't work if we split the app into multiple jobs. The job name and id should be set up correctly in the JobNode.

Reply:
will do